docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,243 @@
+        const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+        pivotTable.dataHierarchies.load("no-properties-needed");
+        await context.sync();
+
+        pivotTable.dataHierarchies.items[0].summarizeBy = Excel.AggregationFunction.average;
+        pivotTable.dataHierarchies.items[1].summarizeBy = Excel.AggregationFunction.average;
+        await context.sync();
+    });
+```
+
+## PivotTable layouts
+
+A PivotTable layout defines the placement of hierarchies and their data. You access the layout to determine the ranges where data is stored. 
+The following diagram shows which layout function calls correspond to which ranges of the PivotTable.
+
+![A diagram showing which sections of a PivotTable are returned by the layout's get range functions.](../images/excel-pivots-layout-breakdown.png)
+
+The following code demonstrates how to get the last row of the PivotTable data by going through the layout. Those values are then summed together for a grand total.
+
+
+```ts
+    await Excel.run(async (context) => {
+        const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+        const range = pivotTable.layout.getDataBodyRange();
+        const grandTotalRange = range.getLastRow();
+        grandTotalRange.load();
+        await context.sync();
+		
+		const masterTotalRange = context.workbook.worksheets.getActiveWorksheet().getRange("B27:C27");
+        masterTotalRange.formulas = [["All Crates", "=SUM(" + grandTotalRange.address + ")"]];
+        await context.sync();
+    });
+```
+
+PivotTables have three layout styles: Compact, Outline, and Tabular. We’ve seen the compact style in the previous examples. 
+The following examples use the outline and tabular styles, respectively. The code sample shows how to cycle between the different layouts.
+
+### Outline layout
+![A PivotTable using the outline layout.](../images/excel-pivots-outline-layout.png)
+
+### Tabular layout
+![A PivotTable usingthe tabular layout.](../images/excel-pivots-tabular-layout.png)
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+	pivotTable.layout.load("layoutType");
+	await context.sync();
+	if (pivotTable.layout.layoutType === "Compact") {
+		pivotTable.layout.layoutType = "Outline";
+	} else if (pivotTable.layout.layoutType === "Outline") {
+		pivotTable.layout.layoutType = "Tabular";
+	} else {
+		pivotTable.layout.layoutType = "Compact";
+	}
+	
+	await context.sync();
+});
+```
+
+## Change hierarchy names
+
+Hierarchy fields are editable. The following code demonstrates how to change the displayed names of two data hierarchies.
+
+```ts
+await Excel.run(async (context) => {
+	const dataHierarchies = context.workbook.worksheets.getActiveWorksheet()
+		.pivotTables.getItem("Farm Sales").dataHierarchies;
+	dataHierarchies.load("name");
```

Comment:
The line
`dataHierarchies.load("name"); `
should be
`dataHierarchies.load("no-properties-needed");` 

You are _writing_ to the `name `properties after the sync, but you don't need to load a property to write to it, only if you are _reading_ it. (The logic of our API's `load()` method is very hard to learn. Not a good experience for our developers.)